### PR TITLE
Add option for docker runtime to provide default volumes

### LINKED ISSUE #993: Allow volumes to be mounted for all plugin containers
### Clear and concise description of the problem

I try to use Woodpecker with a non publicly reachable (Gitea) repository. It's using a certificate not verifyable via standard (i.e. included in standard images) CAs. I can mount /etc/ssl/certs into my woodpecker ui/agent containers, but this doesn't help with any plugins - including the one used to clone the repo, which then fails due to an SSL error.

### Suggested solution

Allowing to configure a common mount for all plugin containers (maybe optionally RO) would make it possible to provide such system settings/files without requiring an admin to label every single repo as trusted and dealing with the security implications of mounting arbitrary paths via CI YAML.
This might even help with caching as as mentioned in #758 or provide other common resources.

### Alternative

* making all containers trusted and manually configure the clone step with volume mounts - This is terrible for security and quite the hassle on top of it.
* building custom plugin containers with updated CAs - This will still require every repo to configure a custom clone step and requires building (and keeping up to date) of plugin containers as well as a private registry to those container images.

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #993

~~Not tested yet, will test once PR docker image is built~~ Tested and works just fine 😊

### CODE REVIEW

File: pipeline/backend/docker/docker.go [hunk 1]
```diff
@@ -59,6 +60,18 @@ func (e *docker) Load() error {
 
 	e.network = os.Getenv("WOODPECKER_BACKEND_DOCKER_NETWORK")
 
+	volumes := strings.Split(os.Getenv("WOODPECKER_BACKEND_DOCKER_VOLUMES"), ",")
```
[the remark this answers is not in the document]
Comment (lafriks) [reply pipeline/backend/docker/docker.go:63]: In example it's single volume/bind, you can add multiple such volumes and between them you would have to put comma

[the remark this answers is not in the document]
Comment (lafriks) [reply pipeline/backend/docker/docker.go:63]: done

[the remark this answers is not in the document]
Comment (anbraten) [reply pipeline/backend/docker/docker.go:63]: We would need to allow escaping `:` and `,`? Can you use those in file names? 🤔

[the remark this answers is not in the document]
Comment (lafriks) [reply pipeline/backend/docker/docker.go:63]: I don't think they should be needed as it won't be supported in pipeline volumes also

**Comment by lafriks** [comment]
Tested and confirm working

File: pipeline/backend/docker/docker.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lafriks) [reply pipeline/backend/docker/docker.go:63]: I don't think `splitVolumeParts` supports escaping `:` that is used for volumes in pipelines

**Comment by anbraten** [comment]
In case the variable is not set this wont allow to run pipelines anymore: `Error response from daemon: invalid volume specification: ''`

**Pull request merged**